### ISSUE TITLE
FP nonstandard_macro_braces with println{} and successive code 

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#9913]
### Summary

.

### Lint Name

nonstandard_macro_braces

### Reproducer

I tried this code:

```rust
fn main() -> Result<(), std::io::Error> {
    println!{""}
    [0]; // separate statement, not indexing into the result of println.
    Ok(())
}
```

I saw this happen:
` cargo clippy --fix -- -Aclippy::all -Wclippy::nonstandard_macro_braces`
```
    Checking clpy v0.1.0 (/tmp/clpy)
warning: failed to automatically apply fixes suggested by rustc to crate `clpy`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0608]: cannot index into a value of type `()`
 --> src/main.rs:2:5
  |
2 | /     println!("")
3 | |     [0]; // separate statement, not indexing into the result of println.
  | |_______^
  |
help: to access tuple elements, use
  |
2 ~     {
3 +         $crate::io::_print($crate::format_args_nl!($($arg)*));
4 ~     }.0; // separate statement, not indexing into the result of println.
  |

error: aborting due to previous error

For more information about this error, try `rustc --explain E0608`.
Original diagnostics will follow.

warning: use of irregular braces for `println!` macro
 --> src/main.rs:2:5
  |
2 |     println!{""}
  |     ^^^^^^^^^^^^ help: consider writing: `println!("")`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#nonstandard_macro_braces
  = note: requested on the command line with `-W clippy::nonstandard-macro-braces`

warning: `clpy` (bin "clpy") generated 1 warning (run `cargo fix --bin "clpy"` to apply 1 suggestion)
warning: `clpy` (bin "clpy" test) generated 1 warning (1 duplicate)
    Finished dev [unoptimized + debuginfo] target(s) in 0.51s
```

### Version

```text
rustc 1.67.0-nightly (c5d82ed7a 2022-11-19)
binary: rustc
commit-hash: c5d82ed7a4ad94a538bb87e5016e7d5ce0bd434b
commit-date: 2022-11-19
host: x86_64-unknown-linux-gnu
release: 1.67.0-nightly
LLVM version: 15.0.4
```
